server/sonar-web/src/main/js/apps/overview/main/CodeSmells.js

```diff
@@ -0,0 +1,157 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import moment from 'moment';
+import React from 'react';
+
+import enhance from './enhance';
+import { IssuesLink } from '../../../components/shared/issues-link';
+import { getMetricName } from '../helpers/metrics';
+import { translate, translateWithParameters } from '../../../helpers/l10n';
+import { formatMeasure, isDiffMetric } from '../../../helpers/measures';
+
+class CodeSmells extends React.Component {
+  renderHeader () {
+    return this.props.renderHeader(
+        'Maintainability',
+        translate('metric.code_smells.name'));
+  }
+
+  renderDebt (metric, type) {
+    const { measures, component } = this.props;
+    const measure = measures.find(measure => measure.metric.key === metric);
+    const value = this.props.getValue(measure);
+    const params = { resolved: 'false', facetMode: 'effort', types: type };
+
+    if (isDiffMetric(metric)) {
+      Object.assign(params, { sinceLeakPeriod: 'true' });
+    }
+
+    const formattedSnapshotDate = moment(component.snapshotDate).format('LLL');
+    const tooltip = translateWithParameters('widget.as_calculated_on_x', formattedSnapshotDate);
+
+    return (
+        <IssuesLink component={component.key} params={params}>
+          <span title={tooltip} data-toggle="tooltip">
+            {formatMeasure(value, 'SHORT_WORK_DUR')}
+          </span>
+        </IssuesLink>
+    );
+  }
+
+  renderTimelineStartDate () {
+    const momentDate = moment(this.props.historyStartDate);
+    const fromNow = momentDate.fromNow();
+    return (
+        <span className="overview-domain-timeline-date">
+          {translateWithParameters('overview.started_x', fromNow)}
+        </span>
+    );
+  }
+
+  renderTimeline (range, displayDate) {
+    return this.props.renderTimeline(
+        'sqale_index',
+        range,
+        displayDate ? this.renderTimelineStartDate(range) : null);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) "this.renderTimelineStartDate" expects "0" arguments, but "1" were provided. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=javascript%3AS930)